src/registrar/tests/common.py

```diff
@@ -824,9 +826,77 @@ def mockSend(self, _request, cleaned):
                 raise RegistryError(
                     code=ErrorCode.OBJECT_ASSOCIATION_PROHIBITS_OPERATION
                 )
-
         return MagicMock(res_data=[self.mockDataInfoHosts])
 
+    def mockUpdateDomainCommands(self, _request, cleaned):
+        if getattr(_request, "name", None) == "dnssec-invalid.gov":
+            raise RegistryError(code=ErrorCode.PARAMETER_VALUE_RANGE_ERROR)
+        else:
+            return MagicMock(
+                res_data=[self.mockDataHostChange],
+                code=ErrorCode.COMMAND_COMPLETED_SUCCESSFULLY,
+            )
+
+    def mockInfoDomainCommands(self, _request, cleaned):
+        request_name = getattr(_request, "name", None)
+
+        # Define a dictionary to map request names to data and extension values
+        request_mappings = {
+            "security.gov": (self.infoDomainNoContact, None),
+            "dnssec-dsdata.gov": (
+                self.mockDataInfoDomain,
+                self.dnssecExtensionWithDsData,
+            ),
+            "dnssec-multdsdata.gov": (
+                self.mockDataInfoDomain,
+                self.dnssecExtensionWithMultDsData,
+            ),
+            "dnssec-keydata.gov": (
+                self.mockDataInfoDomain,
+                self.dnssecExtensionWithKeyData,
+            ),
+            "dnssec-none.gov": (self.mockDataInfoDomain, None),
+            "my-nameserver.gov": (
+                self.infoDomainTwoHosts
+                if self.mockedSendFunction.call_count == 5
```

Comment:
**comment** I do feel this makes it harder to set conditionals for a given domain to return a mock value based on call count or other conditionals.

Reply:
linter made us do it ...